File: 백준/Gold/9935. 문자열 폭발/문자열 폭발.java
import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStreamReader;
import java.util.Stack;

public class Main {

    public static void main(String[] args) throws IOException {

        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        String originalStrings = br.readLine();
        char[] bombString = br.readLine().toCharArray();

        final int BOMB_STRING_LENGTH = bombString.length;

        Stack<Character> stack = new Stack<>();
        for (char c : originalStrings.toCharArray()) {

            stack.push(c);

            boolean flag = true;
            if (stack.size() >= BOMB_STRING_LENGTH) {
                for (int i = 0; i < BOMB_STRING_LENGTH; i++) {
                    if (stack.get(stack.size() - 1 - i) != bombString[BOMB_STRING_LENGTH - 1 - i]) {
                        flag = false;
                        break;
                    }
                } // for 끝
            }

            // stack 안에 폭탄 문자열이 포함되어 있다면
            if (flag && stack.size() >= BOMB_STRING_LENGTH) {
                for (int i = 0; i < BOMB_STRING_LENGTH; i++) {
                    stack.pop();
                }
            }
        } // for 끝

        if (stack.isEmpty()) {
            System.out.println("FRULA");
        } else {
            StringBuilder sb = new StringBuilder();
            for (char c : stack) {
                sb.append(c);
            }
            System.out.println(sb);
        }

    }
}

File: 백준/Gold/9935. 문자열 폭발/문자열 폭발.py
import sys
from collections import deque

input = sys.stdin.readline
string = list(input().rstrip())
bomb = list(input().rstrip())

stack = deque()
bomb_length = len(bomb)
string_length = len(string)


for i in range(string_length):
    stack.append(string[i])
    # 폭탄 문자열 확인
    while True:
        flag = False

        if string_length < bomb_length:  # bomb_length가 더 길다면
            break

        for i in range(bomb_length):
            if stack[len(stack) - 1 - i] != bomb[bomb_length - 1 - i]:
                flag = True
                break

        if flag:
            break

        # 위의 폭탄 문자열을 확인해봤을 때 여기까지 넘어오면 폭탄 문자열이 포함되어 있다는 뜻
        # 폭탄 문자열 수만큼 스택 팝
        for _ in range(bomb_length):
            stack.pop()

        break


if not stack:
    print('FRULA')
else:
    print("".join(stack))